idevice/src/services/misagent.rs

```diff
@@ -2,200 +2,316 @@
+        match err {
+            MisagentError::Success => IdeviceError::UnexpectedResponse, // Shouldn't happen
+            MisagentError::InvalidArgument => IdeviceError::FfiInvalidArg,
+            MisagentError::PlistError => IdeviceError::UnexpectedResponse,
+            MisagentError::ConnectionFailed => IdeviceError::NoEstablishedConnection,
+            MisagentError::RequestFailed => IdeviceError::MisagentFailure,
+            MisagentError::UnknownError => IdeviceError::MisagentFailure,
+        }
     }
 }
 
+/// Client for interacting with the iOS misagent service
+///
+/// Handles provisioning profile installation, listing, and removal operations.
+/// Uses traditional lockdown service connection (not RSD) for maximum compatibility.
+pub struct MisagentClient {
+    /// The underlying device connection with established misagent service
+    pub idevice: Idevice,
+    /// Last error code from the service
+    pub last_error: i32,
+}
+
 impl IdeviceService for MisagentClient {
     /// Returns the misagent service name as registered with lockdownd
     fn service_name() -> std::borrow::Cow<'static, str> {
         obf!("com.apple.misagent")
     }
 
+    /// Establishes a connection to the misagent service
+    ///
+    /// # Arguments
+    /// * `provider` - Device connection provider
+    ///
+    /// # Returns
+    /// A connected `MisagentClient` instance
+    ///
+    /// # Errors
+    /// Returns `IdeviceError` if any step of the connection process fails
+    ///
+    /// # Process
+    /// For USB connections:
+    /// 1. Connects to lockdownd service
+    /// 2. Starts a lockdown session
+    /// 3. Requests the misagent service port
+    /// 4. Establishes connection to the service port
+    /// 5. Optionally starts TLS if required by service
+    ///
+    /// For network connections (TCP):
+    /// 1. Connects to lockdownd service
+    /// 2. Starts a lockdown session
+    /// 3. Performs RSD check-in to get proper entitlements
+    /// 4. Requests the misagent service port
+    /// 5. Establishes connection to the service port
+    /// 6. Starts TLS session
+    async fn connect(
+        provider: &dyn crate::provider::IdeviceProvider,
+    ) -> Result<Self, IdeviceError> {
+        // Check if this is a network connection by examining the provider label
+        // Network connections typically have IP addresses in their labels
+        let provider_label = provider.label();
+        let is_network_connection = provider_label.contains('.') && 
+            provider_label.chars().any(|c| c.is_ascii_digit());
+        
+        debug!("Connecting to misagent service (network: {})", is_network_connection);
+        
+        let mut lockdown = LockdownClient::connect(provider).await?;
+        lockdown
+            .start_session(&provider.get_pairing_file().await?)
+            .await?;
+            
+        // For network connections, perform RSD check-in to get proper entitlements
+        if is_network_connection {
```

Comment:
This is not correct, RSD checkin in only required for RSD connections, not lockdown connections.